controllers/application/deployment.go

```diff
@@ -24,7 +24,13 @@ const (
 	AnnotationKeyLinkPrefix = "link.argocd.argoproj.io/external-link"
 )
 
-var deploymentLog = ctrl.Log.WithName("deployment")
+var (
+	deploymentLog         = ctrl.Log.WithName("deployment")
+	defaultPodAnnotations = map[string]string{
+		"argocd.argoproj.io/sync-options": "Prune=false",
+		"prometheus.io/scrape":            "true",
```

Comment:
Er scrape-annotasjonen egentlig nødvendig? ServiceMonitor ser vel ikke på denne annotasjonen når den scraper metrikker, og setter vi denne til true risikerer vi vel at metrikkene blir plukket opp av Googles egen monitorering?

Reply:
Denne (+ de andre `prometheus.io`-annotasjonene) er påkrevd for at istio-proxy skal kunne scrape pod'en. Omtalt løst [her](https://istio.io/latest/docs/ops/integrations/prometheus/#option-1-metrics-merging) og [diskutert her](https://discuss.istio.io/t/enableprometheusmerge-and-istio-prometheus-annotations/7513). Disse annotasjonene blir til en miljøvariabel som blir injected som config til istio-proxy. Eksempelvis:
```sh
ISTIO_PROMETHEUS_ANNOTATIONS='{"scrape":"true","path":"/actuator/prometheus","port":"8184"}'
```

Reply:
Aha, da forstår jeg. Never mind :sweat_smile: